db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/DefaultPredicateSelectivity.java

```diff
@@ -57,4 +57,14 @@ public double getSelectivity() throws StandardException {
         }
         return selectivity;
     }
+
+    @Override
+    public boolean shouldApplySelectivity() {
+        return
+            getNumReferencedTables() < 2             ||
+            baseTable.getCurrentAccessPath() == null ||
```

Comment:
I don't know, this is more like defensive programming so that no errors occur in any state of FromTable.  For example resetAccessPaths() can set currentAccessPath to null.  If currentAccessPath could never be set to null then this condition is not needed.  But if the method assumes currentAccessPath must be non-null, it places a precondition upon calling this method, which should be documented.  Adding the check just makes life easier because there is no precondition for calling this function.

Reply:
That makes sense. I debugged it a little bit and it seems `getCurrentAccessPath()` can indeed return null. Not sure if it can ever happen during cost estimation, but it seems when the best conglomerate is chosen, current is then set to null. Anyway, I agree that it safer. All good.